packages/domains-table/src/domains-table/domains-table-site-cell.tsx

```diff
@@ -0,0 +1,57 @@
+import { DomainData, SiteDetails } from '@automattic/data-stores';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+
+interface DomainsTableSiteCellProps {
+	site?: Pick< SiteDetails, 'ID' | 'name' >;
+	currentDomainData?: DomainData;
+	siteSlug: string;
+}
+
+export const DomainsTableSiteCell = ( {
+	site,
+	currentDomainData,
+	siteSlug,
+}: DomainsTableSiteCellProps ) => {
+	const { __ } = useI18n();
+
+	if ( ! site || ! currentDomainData ) {
+		return null;
+	}
+
+	if ( currentDomainData.current_user_can_create_site_from_domain_only ) {
+		return createInterpolateElement(
+			/* translators: ariaHidden means that the component will be skipped by screen readers. */
+			__(
+				'<create>Create</create> <ariaHidden>or</ariaHidden> <connect>connect</connect> <ariaHidden>a site</ariaHidden>'
+			),
+			{
+				create: (
+					<a
+						href={ domainOnlySiteCreationLink( siteSlug, site.ID ) }
+						aria-label={ __( 'Create a site for this domain' ) }
+					/>
+				),
+				connect: (
+					<a
+						href={ domainManagementTransferToOtherSiteLink( siteSlug, currentDomainData.domain ) }
+						aria-label={ __( 'Connect this domain to an existing site' ) }
+					/>
+				),
+				ariaHidden: <span aria-hidden={ true } />,
+			}
+		);
+	}
+
+	return site.name ?? '-';
+};
+
+export function domainOnlySiteCreationLink( siteSlug: string, siteId: number ) {
+	return `/start/site-selected/?siteSlug=${ encodeURIComponent(
+		siteSlug
+	) }&siteId=${ encodeURIComponent( siteId ) }`;
```

Comment:
using `addQueryArg` from `@wordpress/url` deals with the encoding for us.

Reply:
Yeah, I just copied this code over!

Reply:
👍 follow up PR https://github.com/Automattic/wp-calypso/pull/81120